components/eam/src/physics/cam/check_energy.F90

```diff
@@ -1291,6 +1111,124 @@ subroutine check_prect(state, tend, name, nstep, ztodt, prect)
 
   end subroutine check_prect
 
+!====================================================================
+
+  subroutine energy_helper_eam_def(u,v,T,q,ps,pdel,phis, &
+                                   ke,se,wv,wl,wi,wr,ws,te,tw, &     
+                                   ncol,teloc,psterm)
+
+!state vars are of size psetcols,pver, so, not exactly correct
+    real(r8), intent(in) :: u(pcols,pver) 
+    real(r8), intent(in) :: v(pcols,pver) 
+    real(r8), intent(in) :: T(pcols,pver) 
+    real(r8), intent(in) :: q(pcols,pver,pcnst) 
+    real(r8), intent(in) :: ps(pcols) 
+    real(r8), intent(in) :: pdel(pcols,pver) 
+    real(r8), intent(in) :: phis(pcols) 
+
+
+    real(r8), intent(inout) :: ke(ncol)     ! vertical integral of kinetic energy
+    real(r8), intent(inout) :: se(ncol)     ! vertical integral of static energy
+    real(r8), intent(inout) :: wv(ncol)     ! vertical integral of water (vapor)
+    real(r8), intent(inout) :: wl(ncol)     ! vertical integral of water (liquid)
+    real(r8), intent(inout) :: wi(ncol)     ! vertical integral of water (ice)
+    real(r8), intent(inout) :: te(ncol)     ! vertical integral of total energy
+    real(r8), intent(inout) :: tw(ncol)     ! vertical integral of total water
+    real(r8), intent(inout) :: wr(ncol)     ! vertical integral of rain
+    real(r8), intent(inout) :: ws(ncol)     ! vertical integral of snow
+
+! do not use in this version
```

Comment:
This comment is a little confusing.

Reply:
This is for future functionality that is not in this PR.